src/components/AbstractModal/AbstractModal.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import Modal from 'react-modal';
+
+const AbstractModal = ({ modalId, className, required, modalIsOpen, children }) => {
+    return (
+        <Modal 
+            id={ modalId } 
+            className={ className }
+            shouldCloseOnOverlayClick={ !required }
+            shouldCloseOnEsc={ !required }
+            isOpen={ modalIsOpen }
+        >
+            { children }
```

Comment:
optional: add a close button here. you will need to pass it an onClick function that will set modalIsOpen to false. 

In order to make this work, we will also need to save modalIsOpen and setModalIsOpen in state and then pass both of those as props to `<AbstractModal>` where ever `<AbstractModal>` is rendered. 

The other option is to have each modal supply its own close button. In that case you don't need to add anything here! 

I think having it rendered here will help to make our modals look the same everywhere in our app, but again, it can be done another way!